pkg/apis/events/v1alpha1/cloudpubsubsource_defaults.go

```diff
@@ -32,13 +34,14 @@ const (
 )
 
 func (ps *CloudPubSubSource) SetDefaults(ctx context.Context) {
+	ctx = apis.WithinParent(ctx, ps.ObjectMeta)
```

Comment:
Question: just wondering what will be the behavior difference between adding this `withParent` and didn't add it.

Reply:
If it is not there, then when we try to default on the inside, using `apis.ParentMeta(ctx).Namespace`, we always get the empty string namespace, which means we would use the cluster level defaults, not the namespace level defaults.